aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -366,6 +366,28 @@ Tensor& logical_not_out(Tensor& result, const Tensor& self) {
   return result;
 }
 
+Tensor& signbit_out(Tensor& result, const Tensor& self) {
```

Comment:
Signbit is a really interesting function. Instead of implementing this using a new TensorIterator kernel, can this be implemented as a test for whether the input tensor is less than zero or not? 

That test won't work for bool, but for bool you can either error or maybe fill the result tensor with false?

Also: should this function check if the input is a complex tensor and throw an error message if it is? Something like, "complex tensors are not supported by signbit"?

Reply:
I actually had a good discussion on another PR about using TensorIterator vs. existing functions. If you're more comfortable using TensorIterator (and the work is already done, after all) then that's OK.

Reply:
> Signbit is a really interesting function. Instead of implementing this using a new TensorIterator kernel, can this be implemented as a test for whether the input tensor is less than zero or not?
> 
> That test won't work for bool, but for bool you can either error or maybe fill the result tensor with false?
> 
> Also: should this function check if the input is a complex tensor and throw an error message if it is? Something like, "complex tensors are not supported by signbit"?

Thanks for remind! The special case for bool and the complex tensor check have been added.

Reply:
> I actually had a good discussion on another PR about using TensorIterator vs. existing functions. If you're more comfortable using TensorIterator (and the work is already done, after all) then that's OK.

Yep, Just like what discussed in #41588, I need use separate TensorIterator kernels for `output=` supporting.